akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/PersistentBehaviors.scala

```diff
@@ -211,14 +213,24 @@ class PersistentBehavior[Command, Event, State](
     copy(recoveryCompleted = callback)
 
   /**
-   * FIXME snapshots are not implemented yet, this is only an API placeholder
+   * Initiates a snapshot if the given function returns true.
+   * When persisting multiple events at once the snapshot is triggered after all the events have
+   * been persisted.
+   *
+   * `predicate` receives the State, Event and the sequenceNr used for the Event
    */
-  def snapshotOnState(predicate: State ⇒ Boolean): PersistentBehavior[Command, Event, State] = ???
+  def snapshotOn(predicate: (State, Event, Long) ⇒ Boolean): PersistentBehavior[Command, Event, State] =
+    copy(snapshotOn = predicate)
 
   /**
-   * FIXME snapshots are not implemented yet, this is only an API placeholder
+   * Snapshot every N events
+   *
+   * `numberOfEvents` should be greater than 0
    */
-  def snapshotOn(predicate: (State, Event) ⇒ Boolean): PersistentBehavior[Command, Event, State] = ???
+  def snapshotEvery(numberOfEvents: Long): PersistentBehavior[Command, Event, State] = {
+    require(numberOfEvents > 0, s"numberOfEvents should be positive: Was $numberOfEvents")
```

Comment:
maybe we can also log a warning for 1, since that usually means people should not be using persistence in the first place anyway